pkg/logs/input/file/scanner.go

```diff
@@ -155,12 +160,16 @@ func (s *Scanner) scan() {
 
 // addSource keeps track of the new source and launch new tailers for this source.
 func (s *Scanner) addSource(source *config.LogSource) {
+	s.Lock()
```

Comment:
forget about my previous comment, the Mutex is also used for the `activeSources` slice.